frontend/javascripts/dashboard/explorative_annotations_view.tsx

```diff
@@ -145,6 +148,24 @@ class ExplorativeAnnotationsView extends React.PureComponent<Props, State> {
     }
   };
 
+  updateTracingInLocalState = (
+    tracing: APIAnnotationInfo,
+    callback: (arg0: APIAnnotationInfo) => APIAnnotationInfo,
+  ) => {
+    const tracings = this.getCurrentTracings();
+    let updatedTracing = tracing;
+    const newTracings = tracings.map((currentTracing) => {
+      if (currentTracing.id !== tracing.id) {
+        return currentTracing;
+      } else {
+        updatedTracing = callback(currentTracing);
+        return updatedTracing;
+      }
+    });
+    this.setModeState({ tracings: newTracings }, this.state.shouldShowArchivedTracings);
+    return updatedTracing;
```

Comment:
I think, the function does not need to return anything (not used anywhere, I think). then, you can remove the local `updatedTracing` variable etc.